x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MigrateAction.java

```diff
@@ -130,11 +129,8 @@ public boolean isSafeAction() {
         }
     }
 
-    static boolean skipMigrateAction(String phase, IndexMetadata indexMetadata) {
-        // if the index is a searchable snapshot we skip the migrate action (as mounting an index as searchable snapshot
-        // configures the tier allocation preference), unless we're in the frozen phase
-        return (indexMetadata.getSettings().get(LifecycleSettings.SNAPSHOT_INDEX_NAME) != null)
-            && (phase.equals(FROZEN_PHASE) == false);
+    private static boolean skipMigrateAction(IndexMetadata indexMetadata) {
```

Comment:
Can you add a comment about why we only skip for partially-mounted indices here?

Reply:
I believe this method is quite shallow so we should drop it altogether? I believe its usage should be replaced by 
```
SearchableSnapshotsConstants.isPartialSearchableSnapshotIndex(indexMetadata.getSettings());
```
with the appropriate comment about why we only skip in this particular case

Reply:
Inlined in 6bd08fec59b301ff63794cbcb443e008016b119b, commented in 8695b510e32c1972d76492052adf5e680970f10d.